api/transactions.go

```diff
@@ -73,9 +73,9 @@ func toApiTransaction(transaction *transactions.Transaction) *Transaction {
 		preimage = transaction.Preimage
 	}
 
-	var metadata interface{}
-	if transaction.Metadata != "" {
-		jsonErr := json.Unmarshal([]byte(transaction.Metadata), &metadata)
+	var metadata *Metadata
```

Comment:
I don't think this should be a pointer - then you use `&metadata` which is basically getting a pointer to a pointer